### DISCUSSION TITLE
Why does Magick++-config repeat compile definitions?

### DISCUSSION THREAD

**Discussion by spinicist** [ImageMagick/ImageMagick#7195] [Help]
### ImageMagick version

6.9.11 & 7.1.1

### Operating system, version and so on

MacOS

### Description

This is the result of running `Magick++-config --cxxflags` on my system, both with version 6.9 and 7.1:
```
-DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -I/opt/local/include/ImageMagick-6```

I'm curious why the two `-D` defines get repeated 3 times? If I remove two of the copies and only keep one my Magick++ application compiles just fine. Thanks!

**Comment by snibgo** [comment]
I don't know why the `-D` definitions are repeated on your system. This is harmless, of course. On my system (Windows 11), they are not repeated.

```
$ Magick++/bin/Magick++-config --cppflags
-I/home/Alan/imdevins7114/include/ImageMagick-7 -fopenmp -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=32 -DMAGICKCORE_CHANNEL_MASK_DEPTH=32
```

**Comment by urban-warrior** [comment]
On our system, we get

```
$ Magick++-config --cxxflags
-I/usr/local/include/ImageMagick-7 -fopenmp -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -DMAGICKCORE_CHANNEL_MASK_DEPTH=64
```

Not sure why you are getting repeats.  However, as mentioned, it does not cause any issues with the build.

**Reply by spinicist** [reply]
Out of interest, which system are you on? I'm wondering if this is specific to Mac. It's only an inconvenience, but it is weird.
